plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -417,7 +418,7 @@ private function is_request( $type ) {
 			case 'cron':
 				return defined( 'DOING_CRON' );
 			case 'frontend':
-				return ( ! is_admin() || defined( 'DOING_AJAX' ) ) && ! defined( 'DOING_CRON' ) && ! $this->is_rest_api_request();
+				return ( ! is_admin() || defined( 'DOING_AJAX' ) ) && ! defined( 'DOING_CRON' );
```

Comment:
Kind of a nitpick, but it might be more accurate to add a separate case for `rest-api` and move `$this->is_rest_api_request()` into it. Also, defensively, it seems like `default:` should be added in a stack with `case 'frontend:` in case the front end case becomes more specific in the future.

Reply:
You're right! Updated in https://github.com/woocommerce/woocommerce/pull/40648/commits/b43a8cfc47d038b0c0c87f4873f69a70b0fc36cb.

- I restored the original test for `'frontend'` which, as you say, is more accurate.
- I initially added a case for `'rest-api'` ... then stripped it out in https://github.com/woocommerce/woocommerce/pull/40648/commits/9425dd049d81d38c76a18618144c7f1487852b77: this is a private method, so there are no external users, and internally we can simply use existing method `is_rest_api_request`.
- I did not make `'frontend'` the default clause. Though I definitely think there is a case for adding a default that returns false, it would seem risky to support a scenario where a type like `is_request( 'badmin' )` evaluates to `is_request( 'frontend' )`.

Reply:
> Though I definitely think there is a case for adding a default that returns false, it would seem risky to support a scenario where a typo like is_request( 'badmin' ) evaluates to is_request( 'frontend' ).

Good point 👍